addons/test_mail/tests/test_ir_attachment.py

```diff
@@ -0,0 +1,76 @@
+from odoo.addons.mail.tests.common import mail_new_test_user
+from odoo.addons.test_mail.tests.common import TestMailCommon
+from odoo.exceptions import AccessError
+
+class TestAttachment(TestMailCommon):
+
+    def test_attachment_forbid_unlink(self):
+        """Check that removing message attachments is prevented on other user's messages."""
+        test_record = self.env['mail.test.simple'].with_context(self._test_context).create({
+            'name': 'Test',
+            'email_from': 'ignasse@example.com',
+        })
+
+        user_admin, user_employee = self.env['res.users'].search([('login', 'in', ['admin', 'employee'])], order='login asc')
+        user_second_employee = mail_new_test_user(
+            self.env,
+            login="employee_second",
+            email="employee_second@example.com",
+            groups="base.group_user,base.group_partner_manager",
+        )
+
+        admin_attachments = self.env['ir.attachment'].with_user(user_admin).create([{
+            'type': 'binary',
+            'name': f'admin_attachment_{n}',
+            'datas': "ABC=",
+        } for n in range(2)])
+        employee_attachments = self.env['ir.attachment'].with_user(user_employee).create([{
+            'type': 'binary',
+            'name': f'employee_attachment_{n}',
+            'datas': "ABC=",
+        } for n in range(3)])
+
+        second_employee_attachments = self.env['ir.attachment'].with_user(user_second_employee).create({
+            'type': 'binary',
+            'name': 'second_employee_attachment',
+            'datas': "ABC=",
+        })
+
+        test_record.with_user(user_admin).message_post(body="Hi", attachment_ids=admin_attachments.ids)
+        test_record.with_user(user_employee).message_post(body="Hello", attachment_ids=employee_attachments.ids)
+        test_record.with_user(user_second_employee).sudo().message_post(body="Hello again", attachment_ids=employee_attachments[2].ids)
```

Comment:
Can't we allow "user_second_employee" to message_post without a sudo?

Would be closer to a real use case.